cmd/ttn-lw-cli/commands/end_devices.go

```diff
@@ -618,6 +623,25 @@ var (
 				return errAddressMismatchEndDevice
 			}
 
+			if extJS {
+				if !config.JoinServerEnabled {
+					return errJoinServerDisabled
+				}
+				js, err := api.Dial(ctx, config.JoinServerGRPCAddress)
+				if err != nil {
+					return err
+				}
+				_, err = ttnpb.NewJsEndDeviceRegistryClient(js).Delete(ctx, &device.EndDeviceIdentifiers)
+				if errors.IsNotFound(err) {
+					logger.WithError(err).Error("Could not delete end device from Join Server")
+				} else if err != nil {
+					return err
+				}
+				config.JoinServerEnabled = false
```

Comment:
What is this for?

Reply:
I think this entire block could use some comments to tell other developers what's going on

Reply:
That's to avoid setting fields on JS